src/main/java/core/basesyntax/library/Injector.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.library;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazs) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+
+        Constructor constructor = clazs.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazs.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+
+                field.setAccessible(true);
+
+                if (field.getType() == BetDao.class) {
+                    if (!BetDaoImpl.class.isAnnotationPresent(Dao.class)) {
```

Comment:
we can combine if blocks in 1 with logic operator &&. And use if fieldType == BetDao.class && BetDaoImpl.class.isAnnotation present without "!" and then set. else if with userDao class and else throw new exception

Reply:
not agree :) I think need check Type and after that need the checking the cases for this type only

Reply:
Yes, you right. But is check annotation in case ( field.getType() == BetDao.class && BetDaoImpl.class.isAnnotationPresent(Dao.class)) not for type BetDao? We can avoid redundant blocks if and duplicate for throw exception. So we have with &&: if type not betdao - false(right part will don't check) otherwise check. Is it equal for 2 ifs? Also i think is better check for true and then make some actions.

Reply:
@andriylagun Victoria wants to check a special case when field type is ok, but the implementation does not have `@Dao` annotation
she has 2 different exceptions, that's why in her case it is alright not to use `&&`

Reply:
:)